### PR TITLE
Biomass Reclaimer cleanup

### PR DESCRIPTION
**Pull request by 0x6273**
#11545 got merged while I was reviewing it. This PR addresses my review comments, plus some other minor changes.

- Document all the properties.
- Data fields for how much the part ratings affect speed/yield, instead of hardcoded values.
- The timers now count down instead of up (gets rid of `Accumulator` field).
- Make `BloodReagent` nullable.
- Make `CurrentExpectedYield` uint instead of float.
- More `ViewVariables` attributes and made some existing ones RW.
- Decouple speed and yield so they no longer affect the other, instead both are based on mass.
- Change formula for yield to be the same as for speed: multiplier^(partRating-1)

### CODE REVIEW

File: Content.Server/Medical/BiomassReclaimer/BiomassReclaimerComponent.cs [hunk 1]
```diff
@@ -10,39 +10,54 @@ public sealed class BiomassReclaimerComponent : Component
...
+        /// Entities that can be randomly spawned while processing a mob.
+        /// </summary>
         public List<EntitySpawnEntry> SpawnedEntities = new();
 
         /// <summary>
         /// How many units of biomass it produces for each unit of mass.
         /// </summary>
         [ViewVariables(VVAccess.ReadWrite)]
-        public float YieldPerUnitMass = 0.4f;
+        public float YieldPerUnitMass = default!;
```
Comment (metalgearsloth) [inline Content.Server/Medical/BiomassReclaimer/BiomassReclaimerComponent.cs:55]: You don't need to null suppress primitive types.

File: Content.Server/Medical/BiomassReclaimer/BiomassReclaimerSystem.cs [hunk 2]
```diff
@@ -181,6 +179,12 @@ private void OnReclaimCancelled(ReclaimCancelledEvent args)
         }
         private void StartProcessing(EntityUid toProcess, BiomassReclaimerComponent component)
         {
+            if (!TryComp<PhysicsComponent>(toProcess, out var physics))
+            {
+                Logger.Error("Somehow tried to extract biomass from {0}, which has no physics component.", toProcess);
```
[the remark this answers is not in the document]
Comment (0x6273) [reply Content.Server/Medical/BiomassReclaimer/BiomassReclaimerSystem.cs:184]: So I should do something like this instead?
```csharp
PhysicsComponent? physics = null;
if (!Resolve(toProcess, ref physics))
    return;
```

File: Content.Server/Medical/BiomassReclaimer/BiomassReclaimerSystem.cs [hunk 3]
```diff
@@ -181,6 +179,10 @@ private void OnReclaimCancelled(ReclaimCancelledEvent args)
         }
         private void StartProcessing(EntityUid toProcess, BiomassReclaimerComponent component)
         {
+            PhysicsComponent? physics = null;
```
Comment (vulppine) [inline Content.Server/Medical/BiomassReclaimer/BiomassReclaimerSystem.cs:182]: just place this in the function parameters (most of SS14 does that if the function isn't a registered event listener)

**Review by vulppine — CHANGES_REQUESTED** [review]

File: Content.Server/Medical/BiomassReclaimer/BiomassReclaimerSystem.cs [hunk 4]
```diff
@@ -192,20 +194,10 @@ private void StartProcessing(EntityUid toProcess, BiomassReclaimerComponent comp
                 component.SpawnedEntities = butcherableComponent.SpawnedEntities;
             }
 
-            component.CurrentExpectedYield = CalculateYield(toProcess, component);
-            component.CurrentProcessingTime = component.CurrentExpectedYield / component.YieldPerUnitMass * component.ProcessingSpeedMultiplier;
+            component.CurrentExpectedYield = (uint) Math.Max(0, physics.FixturesMass * component.YieldPerUnitMass);
+            component.ProcessingTimer = physics.FixturesMass * component.ProcessingTimePerUnitMass;
             EntityManager.QueueDeleteEntity(toProcess);
```
Comment (vulppine) [inline Content.Server/Medical/BiomassReclaimer/BiomassReclaimerSystem.cs:199]:
```suggestion
            QueueDel(toProcess);
```

**Review by vulppine — APPROVED** [review]

**Review by mirrorcult — APPROVED** [review]

**Pull request merged**